### PR TITLE
add parametrized tests and parallel test execution

### PR DESCRIPTION
**Pull request by OlegChub**

### CODE REVIEW

File: src/main/java/controllers/LaunchController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            case "failed":
+                return launchExecutions.getFailed();
+            case "passed":
+                return launchExecutions.getPassed();
+            case "skipped":
+                return launchExecutions.getSkipped();
+            default:
+                System.out.println("There is no such parameter in launch executions info block");
+        }
+
```
Comment (Fentezi3) [inline src/main/java/controllers/LaunchController.java:55]: Unnecessary line

File: src/main/java/controllers/LaunchController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return -1;
+    }
+
+    public int getLaunchDefectsInfoParameter(String parameterName2) {
```
Comment (Fentezi3) [inline src/main/java/controllers/LaunchController.java]: Why are you using '2' in 'parameterName2'?

**Review by Fentezi3 — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/LaunchController.java (hunk 2 above)
Comment (OlegChub) [reply src/main/java/controllers/LaunchController.java]: Left after debugging. Renamed it.

File: src/main/java/controllers/LaunchController.java (hunk 1 above)
Comment (OlegChub) [reply src/main/java/controllers/LaunchController.java:55]: Do you mean line #55?

**Comment by OlegChub** [comment]
image

File: src/main/java/controllers/LaunchController.java (hunk 1 above)
Comment (Fentezi3) [reply src/main/java/controllers/LaunchController.java:55]: yes

File: pom.xml [hunk 3]
```diff
@@ -107,6 +115,12 @@
                     <systemPropertyVariables>
                         <allure.results.directory>${project.build.directory}/allure-results</allure.results.directory>
                     </systemPropertyVariables>
+                    <!--  Source for TestNG running     -->
+<!--                    <suiteXmlFiles>-->
+<!--                        <suiteXmlFile>-->
+<!--                            src\test\resources\testng.xml-->
+<!--                        </suiteXmlFile>-->
+<!--                    </suiteXmlFiles>-->
```
Comment (Fentezi3) [inline pom.xml]: why did you commite this part of the code?

**Review by Fentezi3 — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/LaunchController.java (hunk 1 above)
Comment (OlegChub) [reply src/main/java/controllers/LaunchController.java:55]: I did it on purpose to separate the 'switch' block and 'return' of the method. Below is the same thing with 'if' statement

File: pom.xml (hunk 3 above)
Comment (OlegChub) [reply pom.xml]: It belongs to TestNG. And as my main runner on the project is JUnit, so I decided to comment it. But it was unnecessary. Uncommented

File: src/main/java/controllers/LaunchController.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+public class LaunchController {
+    private ApiClient client;
+    private ValidatableResponse validatableResponse;
+    private Response response;
+
+    public LaunchController(ApiClient client) {
+        this.client = client;
+    }
+
+    public void getExactLaunchInfo(int id) {
```
Comment (dsalonka) [inline src/main/java/controllers/LaunchController.java:22]: will you have 'not exact' info?

File: src/main/java/controllers/LaunchController.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        this.validatableResponse = client.execute(new GetLaunchInfo(id));
+        this.response = validatableResponse.extract().response();
```
Comment (dsalonka) [inline src/main/java/controllers/LaunchController.java:24]: validatableResponse used only one, why do you need to create separate private variable? and the same for response

File: src/test/java/testData/DataProviderForTestNG.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package testData;
+
+import org.testng.annotations.DataProvider;
+
+public class DataProviderForTestNG {
+    @DataProvider(name = "testng")
+    public static Object[][] getTestDataForTestNG() {
+        return new Object[][]{
+                {"total", 20},
```
Comment (dsalonka) [inline src/test/java/testData/DataProviderForTestNG.java:9]: what if you need to test 5 launches with ids from 1 to 5? how data provider will look like?  I mean will it be easier to group total, passed, etc as one object launchDetails?

**Comment by OlegChub** [comment]
Allure report
[allure-report.zip](https://github.com/OlegChub/reportPortal/files/9840435/allure-report.zip)

**Comment by OlegChub** [comment]
Allure report
[allure-report.zip](https://github.com/OlegChub/reportPortal/files/9840438/allure-report.zip)

**Comment by OlegChub** [comment]
What was done:
- added alternative test runner (TestNG) 
- implemented data driven tests
- implemented parallelization

**Review by dKatechev — APPROVED** [review]

**Pull request merged**